ground/gcs/src/plugins/uavobjects/uavobjecttemplate.m

```diff
@@ -171,38 +155,86 @@
 		%     Timestamp (2 bytes)
 		%     Data (variable length)
 		%     Checksum (1 byte)
-    
+		
 		% Process header for overo
 		datasizeBufferIdx = bufferIdx + 2;
-		datasizeLength = 2;
 		msgType = buffer(bufferIdx+1) - 128;
 		objID = typecast(buffer(bufferIdx+4:bufferIdx+ 4+4-1), 'uint32');
-        
+		
 		singleInstance = multipleInstanceLookup(multipleInstanceLookup(:,1) == objID, 2);
 		if singleInstance
 			timestamp = double(typecast(buffer(bufferIdx+8:bufferIdx+10-1),'uint16'));
 		else
 			timestamp = double(typecast(buffer(bufferIdx+12:bufferIdx+14-1),'uint16'));
 		end
-        
+		
 		% Advance buffer past header to where data is.  In the case of a
 		% multiple instance object this will be where the timstamp is but
 		% the parsing code will advance by two more.
 		bufferIdx = bufferIdx + 10;
+	elseif onboardLogger
```

Comment:
so I'm puzzled what is going on here. how does onboard logging compare to over? it looks like it has the integrated time stamp but you aren't subtracting 128 from msgType below. Does the onboard logging not set the msgType appropriately?